src/lib/core/fiber.h

```diff
@@ -855,6 +875,73 @@ fiber_name(struct fiber *f)
 	return f->name;
 }
 
+/**
+ * Time since fiber was called.
+ */
+static inline double
+fiber_time_from_call(struct cord *cord)
+{
+	return clock_monotonic() - cord->call_time;
+}
+
+/**
+ * Check if fiber has custom slice.
+ */
+static inline bool
+fiber_has_custom_slice(struct fiber *fiber)
+{
+	return fiber->custom_slice.err != 0.0;
+}
+
+/**
+ * Set current slice.
+ */
+static inline void
+fiber_set_current_slice(struct fiber_slice slice)
+{
+	cord()->current_slice = slice;
+}
+
+/**
+ * Set default slice for fibers without custom slice for current cord.
+ */
+static inline void
+fiber_set_default_slice(struct fiber_slice slice)
+{
+	cord()->default_slice = slice;
+	if (!fiber_has_custom_slice(fiber()))
+		cord()->current_slice = slice;
+}
+
+/**
+ * Set slice to fiber. Must be called in the thread in which the cord
+ * that manages this fiber is located.
+ * Error slice must be greater than 0.
+ */
+static inline void
+fiber_set_slice(struct fiber *fib, struct fiber_slice slice)
+{
+	fib->custom_slice = slice;
+	if (fiber() == fib)
+		cord()->current_slice = slice;
+}
+
+/**
+ * Check if slice is over for current cord.
+ */
+static inline bool
+fiber_slice_is_over(void)
```

Comment:
Let's instead add `fiber_check_slice()` that would set diag in case the slice is exceeded - it seems there's no other action we can do in this case but raise an error. In general, I'd like to have similar C and Lua API to avoid confusion. Ideally, a Lua function should call a C function of the same name.

Reply:
Done